test/functionapp.test.ts

```diff
@@ -0,0 +1,96 @@
+        this.timeout(5 * 60 * 1000);
+        resourceName = fsUtil.getRandomHexString();
+        resourceGroupsToDelete.push(resourceName);
+        const testInputs: string[] = [resourceName, '$(plus) Create new resource group', resourceName, '$(plus) Create new storage account', resourceName, 'East US'];
+        ext.ui = new TestUserInput(testInputs);
+        await vscode.commands.executeCommand('azureFunctions.createFunctionApp', testAccount.getSubscriptionId());
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.ok(createdApp);
+    });
+    test('stopFunctionApp', async function (this: IHookCallbackContext): Promise<void> {
+        this.timeout(5 * 60 * 1000);
+        ext.ui = new TestUserInput([resourceName]);
+        await vscode.commands.executeCommand('azureFunctions.stopFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp1: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.equal(createdApp1.state, 'Stopped', 'Function app State is incorrect.');
+    });
+    test('startFunctionApp', async function (this: IHookCallbackContext): Promise<void> {
+        this.timeout(5 * 60 * 1000);
+        ext.ui = new TestUserInput([resourceName]);
+        await vscode.commands.executeCommand('azureFunctions.startFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp1: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.equal(createdApp1.state, 'Running', 'Function app State is incorrect.');
+    });
+    test('restartFunctionApp', async function (this: IHookCallbackContext): Promise<void> {
+        this.timeout(5 * 60 * 1000);
+        let client: WebSiteManagementClient;
+        let createdApp1: WebSiteManagementModels.Site;
+        ext.ui = new TestUserInput([resourceName]);
+        await vscode.commands.executeCommand('azureFunctions.stopFunctionApp');
+        client = getWebsiteManagementClient(testAccount);
+        createdApp1 = await client.webApps.get(resourceName, resourceName);
+        assert.equal(createdApp1.state, 'Stopped', 'Function app State is incorrect.');
+        //'Restart' is disabled after stopping a function app in Azure portal.
+        ext.ui = new TestUserInput([resourceName, resourceName]);
+        await vscode.commands.executeCommand('azureFunctions.restartFunctionApp');
+        client = getWebsiteManagementClient(testAccount);
+        createdApp1 = await client.webApps.get(resourceName, resourceName);
+        assert.equal(createdApp1.state, 'Running', 'Function app State is incorrect.');
+    });
+    test('deleteFunctionApp', async function (this: IHookCallbackContext): Promise<void> {
```

Comment:
It seems like you kind of duplicated our test case already implemented here: 
https://github.com/Microsoft/vscode-azurefunctions/blob/master/test/createAzureResource.test.ts#L56-L71

Since this is a refactor, could you delete the old test?  We don't need to create/delete a resource twice, but I understand that you needed to create a resource to run the start/stop/restart commands on.